src/main/scala/beam/agentsim/agents/rideHail/allocationManagers/DefaultRideHailResourceAllocationManager.scala

```diff
@@ -14,7 +14,7 @@ class DefaultRideHailResourceAllocationManager extends RideHailResourceAllocatio
   }
 
 // TODO RW/Asif: how to make sure no one ever can call this?
-def allocateVehicles(allocationsDuringReservation: Vector[(VehicleAllocationRequest, Option[VehicleAllocation])]): Vector[(VehicleAllocationRequest, Option[VehicleAllocation])] = {
+def allocateVehicles(allocationsDuringReservation: Vector[(VehicleAllocationRequest, Option[VehicleAllocation])]): IndexedSeq[(VehicleAllocationRequest, Option[VehicleAllocation])] = {
   log.error("batch processing is not implemented for DefaultRideHailResourceAllocationManager")
     //???
   return allocationsDuringReservation
```

Comment:
What's the purpose of the signature change here? Underlying, the data is the same - same for other areas like this

Reply:
Using `IndexedSeq` allows to fill `ArrayBuffer` and return it as result, instead of converting it to vector by calling `toVector`